src/axom/slam/BivariateSet.hpp

```diff
@@ -278,6 +294,56 @@ bool BivariateSet<Set1, Set2>::isValid(bool verboseOutput) const
   return m_set1->isValid(verboseOutput) && m_set2->isValid(verboseOutput);
 }
 
+/*!
+ * \class BivariateSetIterator
+ *
+ * \brief Implements a forward iterator concept on a BivariateSet type.
+ */
+template <typename BivariateSetType>
+struct BivariateSetIterator
+  : public IteratorBase<BivariateSetIterator<BivariateSetType>,
+                        typename BivariateSetType::PositionType>
+{
+public:
+  using IndexType = typename BivariateSetType::PositionType;
+  using BaseType =
+    IteratorBase<BivariateSetIterator<BivariateSetType>, IndexType>;
+  using difference_type = IndexType;
+  using value_type = std::pair<IndexType, IndexType>;
+  using reference = value_type&;
+  using pointer = value_type*;
+  using iterator_category = std::forward_iterator_tag;
+
+  BivariateSetIterator(const BivariateSetType* bset, IndexType flatPos = 0)
+    : BaseType(flatPos)
+    , m_bset(bset)
+  { }
+
+  std::pair<IndexType, IndexType> operator*() const
+  {
+    // Going from flat-to-second index is always free for a StaticRelation.

```

Comment:
Something seems off here.  Maybe you meant
// Going from flat to second-index ...

Reply:
Updated comment.